lib/vix/tensor.ex

```diff
@@ -0,0 +1,45 @@
+defmodule Vix.Tensor do
+  @moduledoc false
+
+  alias Vix.Vips.Image
+
+  defstruct [
+    data: nil,
+    shape: {0, 0, 0},
+    names: [:x, :y, :band],
+    type: {}
+  ]
+
+  def binary_to_tensor(binary, size, %Image{} = image)
+      when is_binary(binary) and is_integer(size) do
+    struct(__MODULE__,
+      data: binary,
+      shape: {Image.width(image), Image.height(image), Image.bands(image)},
+      names: [:x, :y, :band],
+      type: nx_type(image)
+    )
+  end
+
+  # NOTE: This isn't really the right way to determine
+  # the bit sizes of these types. Is there a way to get
+  # this information from an existing NIF call, or do
+  # we need a call to return format bit sizes?
+
+  defp nx_type(image) do
+    word_size = :erlang.system_info(:wordsize)
+
+    case Image.format(image) do
+      :VIPS_FORMAT_UCHAR   -> {:u, 8}
+      :VIPS_FORMAT_CHAR    -> {:s, 8}
+      :VIPS_FORMAT_USHORT  -> {:u, 16}
+      :VIPS_FORMAT_SHORT   -> {:s, 16}
+      :VIPS_FORMAT_UINT    -> {:u, word_size}
```

Comment:
I think in libvips bitsize is fixed, and does not depend on the machine architecture. 

* https://github.com/libvips/pyvips/blob/df462ca09b8d63c8a5a9cd2f21a6564ddcd3ef08/pyvips/vimage.py#L131-L141
* https://github.com/libvips/libvips/blob/6260a37136973f3d231d0e2c28af6190126c664e/libvips/foreign/matlab.c#L147-L156

wdyt?

Reply:
I think that's really great :-)

Reply:
So shall we change this to use `32` and `64` here?